src/merge.c

```diff
@@ -1711,8 +1732,13 @@ int git_merge__iterators(
 
 	git_vector_foreach(&changes, i, conflict) {
 		int resolved = 0;
+		int file_favor = opts.file_favor;
+
+		/* Check for merge options in .gitattributes */
+		if ((error = git_merge__lookup_file_favor(repo, conflict->our_entry.path, &file_favor) < 0))
```

Comment:
Right. I believe union merges should only apply when their is conflict between `ours` and theirs`, so I've tightened up the check so that it will only be enabled when both paths exist and match.